### DISCUSSION TITLE
Calling D365FO tools in Azure functions

### DISCUSSION THREAD

**Discussion by udaymuppana** [d365collaborative/d365fo.tools#618] [Q&A]
Hi All, 
First of all, I would like to thank this community for building such a wonderful library, this is really helping in saving time.

As part of internal automation, we are trying to call D365FO tools from Azure functions. But there is no success so far. 
Does anyone try referring D365FoTools from Azure functions?
We followed Azure functions "Dependency management" settings, I tried both below approaches to resolve the dependency.

- Approach 1. Mentioning D365FOtools dependency in requirements.pds1 (Image1)
- Approach 2. Manually adding nine dependent modules under wwroot\Modules  (Image2)

> **Error message in Azure function: **Get-365LCSApiTokne command was found in the modules 'd365fo.tools', but the module could not be loaded**.**  (Image3)

So ultimately whatever d365foTools functions we are getting the same error. 
Am I missing something in Azure functions settings? Really appreciate any inputs here. 

Thanks & Regards,
Udaya Bhaskar.

**Image1#**
![image](https://user-images.githubusercontent.com/62023339/150239894-8c54bd58-8e5c-4372-9c35-c77453634a90.png)

**Image2#**
![image](https://user-images.githubusercontent.com/62023339/150240338-bc825ea9-abb3-48b8-ae67-40d2e9e86c54.png)

**Image3#**
![image](https://user-images.githubusercontent.com/62023339/150241068-dc5ab964-8472-4fea-a9b1-7b94f13b35d3.png)

**Comment by Splaxi** [comment]
I would look into Azure Automation Accounts instead. Here I know for sure that you can load the d365fo.tools module.

I think you should try with other powershell modules first and make them work inside Azure functions. When it works, we can look into what the difference is.

**Reply by udaymuppana** [reply]
@Splaxi 
Apologies for the delayed response. I was struck by other tasks. 
Azure-hosted pipeline did the job for me. My target was to host d365fo.tools module in some serverless component. 
I could able call d365fo.tools command from Azure-hosted pipeline and powershell inline script task.  

Coming to the earlier challenges:  
- Azure functions still have a lot of dependency complexities, which I could not able to figure out. 
- Azure Automation Accounts, have loaded the d365fo.tools module correctly. but it has some glitches in executing end commands.

I will look into Azure functions and Azure automation tools some other time. 

Thanks.

**Reply by Splaxi** [reply]
Thanks for updating us with your progress!
